lib/trip_dispatcher.rb

```diff
@@ -90,6 +91,56 @@ def load_trips
       trips
     end
 
+    def find_last_trips
+      last_trips = []
+      drivers.each do |driver|
+        unless driver.trips.empty? || driver.trips.first.end_time == nil
+          last_trips << driver.trips.max_by {|trip| trip.end_time.to_i}
+        end
+      end
+      return last_trips
+    end
+
+    def least_recent
+
+      completed_trips = find_last_trips.find_all do |trip|
+        trip.end_time != nil && trip.driver.status == :AVAILABLE
+      end
+      completed_trips.sort! {|trip1, trip2| trip1.end_time <=> trip2.end_time}
+      if completed_trips[0] == nil
+        raise ArgumentError.new("No drivers currently available.")
+      else
+        return completed_trips.first.driver
+      end
+    end
+
+    def select_driver
+        return least_recent
+    end
+
+    def request_trip(passenger_id)
+      driver = select_driver
+      id = trips.length + 1
+
```

Comment:
Line 123 assumes that all the trips are numbered sequentially. This works fine for this project, but if there were any gaps (e.g. a trip had been removed from the middle) it would result in a duplicate ID.